### DISCUSSION TITLE
squishbox with a pine64

### DISCUSSION THREAD

**Discussion by uomodellaforesta** [GeekFunkLabs/fluidpatcher#66] [General]
Hello, my name is Marco. First of all, i should like to express my compliments for the project! But expecially, thank you for sharing it!
I like it too much so i want make my copy of a squishbox as soon as possibile since i need a new sound module for my band 😄 
During this period i can't find any raspberry available to buy, so i decided to use a pine64 as alternative! (i've a 2GB with A64 board)
However i'm newbie with linux and python so i need some suggestions.
Actually i installed armbian OS and i solved a few issues to get analog sound from the 3.5" jack of the board.
Furthermore i installed correctly fluidsynth and python3 and tested them so i can tell that they works correctly.
Finally i moved the "fluidpatcher-master" folder in a usb drive and mounted it in armbian os and i tested "headlesspi.py":
I run the command "python3 headlesspi.py", connected a midi controller (actually a smartphone app with usbmidi) and WOW!! i play the music! 🎶😃 
Now i want to run the "squishbox.py" but i certainly need to make some modification to the project.
So i want to connect the lcd and the buttons/encoder and do other modifications like the capability to run squishbox.py at the start of  the os first boot.

for the first question i think that i need to import a different RPi.GPIO module into stompboxpi.py file and accordingly i need to change the gpio numbers into hw_overlay.py file. Are these modification sufficient to use lcd and buttons/encoder? naturally i will connect the pins relatively to the gpio that i'll configure.

naturally i will keep you up to date :)

Thank you
Marco

**Comment by albedozero** [comment]
Hello Marco,

You're correct - you'll need a modified version of RPi.GPIO ([this forum post](https://forum.pine64.org/showthread.php?tid=723) has a link to one that might work), and you'll need to modify the file assets/hw_overlay.py to refer to the pins as you connect them in your circuit - but that's all you should need to do to make the LCD and buttons work. I'd thought about getting a pine64 board to try this out - let me know how it goes!

**Comment by uomodellaforesta** [comment]
Hello albedozero,
i installed the RPi.GPIO module that you linked to me and it works! i found two code errors and i solved it as you can see at this [link](https://github.com/swkim01/RPi.GPIO-PineA64/issues/14)
i made a test with a custom python code and i discovered that, using this module and BCM configuration, the numbers of GPIO are the same as Raspberry PI :) so i don't needed to change the pin numbers except for the GPIO4 since that in my board it doesn't work: i changed it with GPIO14!

**Reply by albedozero** [reply]
Nicely done and thank you for sharing!
